From 836510517a1cebdd62d67c447ef4dd4309da0459 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Tue, 21 Oct 2014 18:59:14 +0800
Subject: [PATCH 1013/1543] MLK-9723-6: ARM: dts: add
 imx6sx-19x19-arm2-mqs.dts for mqs

commit a56d529c3bd9aeeb733bf5f9594264ccc6e33e55 from
git://git.freescale.com/imx/linux-2.6-imx.git

Initialize dts file for mqs.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
(cherry picked from commit d390f670990a0880cdd0db4271a724a96755b0ea)
---
 arch/arm/boot/dts/imx6sx-19x19-arm2-mqs.dts |   42 +++++++++++++++++++++++++++
 arch/arm/boot/dts/imx6sx-19x19-arm2.dts     |    9 ++++++
 arch/arm/boot/dts/imx6sx.dtsi               |    6 ++++
 3 files changed, 57 insertions(+), 0 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6sx-19x19-arm2-mqs.dts

diff --git a/arch/arm/boot/dts/imx6sx-19x19-arm2-mqs.dts b/arch/arm/boot/dts/imx6sx-19x19-arm2-mqs.dts
new file mode 100644
index 0000000..39eea75
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sx-19x19-arm2-mqs.dts
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-19x19-arm2.dts"
+
+/ {
+	sound-mqs {
+		compatible = "fsl,imx6sx-sdb-mqs",
+				"fsl,imx-audio-mqs";
+		model = "mqs-audio";
+		cpu-dai = <&sai1>;
+		audio-codec = <&mqs>;
+	};
+};
+
+&esai {
+	/* pin conflict with sai */
+	status = "disabled";
+};
+
+&mqs {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_mqs_1>;
+	clocks = <&clks IMX6SX_CLK_SAI1>;
+	clock-names = "mclk";
+	status = "okay";
+};
+
+&sai1 {
+	status = "okay";
+};
+
+&sdma {
+	gpr = <&gpr>;
+	/* SDMA event remap for SAI1 */
+	fsl,sdma-event-remap = <0 15 1>, <0 16 1>;
+};
diff --git a/arch/arm/boot/dts/imx6sx-19x19-arm2.dts b/arch/arm/boot/dts/imx6sx-19x19-arm2.dts
index edf83e6..1af0d4e 100644
--- a/arch/arm/boot/dts/imx6sx-19x19-arm2.dts
+++ b/arch/arm/boot/dts/imx6sx-19x19-arm2.dts
@@ -839,6 +839,15 @@
 		};
 	};
 
+	mqs {
+		pinctrl_mqs_1: mqsgrp-1 {
+			fsl,pins = <
+				MX6SX_PAD_SD2_CLK__MQS_RIGHT 0x80000000
+				MX6SX_PAD_SD2_CMD__MQS_LEFT 0x80000000
+			>;
+		};
+	};
+
 	pwm3 {
 		pinctrl_pwm3_0: pwm3grp-0 {
 			fsl,pins = <
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index ac05197..9f319c5 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -724,6 +724,12 @@
 				fsl,anatop = <&anatop>;
 			};
 
+			mqs: mqs {
+				compatible = "fsl,imx6sx-mqs";
+				gpr = <&gpr>;
+				status = "disabled";
+			};
+
 			snvs: snvs@020cc000 {
 				compatible = "fsl,sec-v4.0-mon", "simple-bus";
 				#address-cells = <1>;
-- 
1.7.5.4

